RivkyKlapholtz/Rebar
# Implementing RebarAPI: MongoDB integration and new Controllers

**Description**:
This pull request introduces significant updates to our backend, focusing primarily on integrating MongoDB and establishing the main controllers for our Rebar application. Detailed overview:

- **MongoDB Configuration and Service**: Configured the MongoDB connection and added `MongoService` to interact with our database collections (`Orders` and `Menu`).

- **Models Introduced**:
  - `Shake`: Represents the shakes in our menu.
  - `OrderItem`: Represents individual items in an order.
  - `Order`: Represents a complete order, including associated items and discounts.
  - `Discount`: Represents discounts that can be applied to orders.

- **API Controllers**:
  - `OrdersController`: CRUD operations for orders.
  - `MenuController`: CRUD operations for menu items.
  - `CheckoutController`: Functions for closing the checkout and fetching daily reports.

- **Swagger Integration**: Integrated Swagger for API documentation and testing purposes.

**How to Test**:
1. Run the application.
2. Use Swagger UI (`/swagger`) to test the API endpoints.
3. For database-related testing, ensure you have MongoDB running and the database `RebarDB` set up.
